src/Symfony/Component/Workflow/Tests/RegistryTest.php

```diff
@@ -64,6 +63,40 @@ public function testGetWithNoMatch()
         $this->assertInstanceOf(Workflow::class, $w1);
         $this->assertSame('workflow1', $w1->getName());
     }
+
+    /**
+     * @group legacy
+     */
+    public function testGetWithSuccessLegacyStrategy()
+    {
+        $registry = new Registry();
+
+        $registry->add(new Workflow(new Definition(array(), array()), $this->getMockBuilder(MarkingStoreInterface::class)->getMock(), $this->getMockBuilder(EventDispatcherInterface::class)->getMock(), 'workflow1'), Subject1::class);
+        $registry->add(new Workflow(new Definition(array(), array()), $this->getMockBuilder(MarkingStoreInterface::class)->getMock(), $this->getMockBuilder(EventDispatcherInterface::class)->getMock(), 'workflow2'), Subject2::class);
+
+        $workflow = $registry->get(new Subject1());
+        $this->assertInstanceOf(Workflow::class, $workflow);
+        $this->assertSame('workflow1', $workflow->getName());
+
+        $workflow = $registry->get(new Subject1(), 'workflow1');
+        $this->assertInstanceOf(Workflow::class, $workflow);
+        $this->assertSame('workflow1', $workflow->getName());
+
+        $workflow = $registry->get(new Subject2(), 'workflow2');
+        $this->assertInstanceOf(Workflow::class, $workflow);
+        $this->assertSame('workflow2', $workflow->getName());
+    }
+
+    private function getSupportStrategy($supportedClassName)
+    {
+        $strategy = $this->getMockBuilder(SupportStrategyInterface::class)->getMock();
+        $strategy->expects($this->any())->method('supports')
+            ->will($this->returnCallback(function ($workflow, $subject) use ($supportedClassName) {
+                return $subject instanceof $supportedClassName;
```

Comment:
Don't know how this is handled in Symfony environment, but as I want to test the registry alone I personally prefer mocking all other collaborators. Should I change it?